src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = BridgeTowerConfig
+    base_model_prefix = "bridgetower"
+    supports_gradient_checkpointing = True
+    _no_split_modules = ["BridgeTowerSelfAttention"]
+
+    def _init_weights(self, module):
+        if isinstance(module, (nn.Linear, nn.Embedding)):
+            module.weight.data.normal_(mean=0.0, std=0.02)
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+        if isinstance(module, nn.Linear) and module.bias is not None:
+            module.bias.data.zero_()
+
+
+@add_start_docstrings(
+    (
+        "The bare BridgeTower Model transformer outputting BridgeTowerModelOutput object without any specific head on"
+        " top."
+    ),
+    BRIDGETOWER_START_DOCSTRING,
+)
+class BridgeTowerModel(BridgeTowerPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+
+        self.tokenizer_config = RobertaConfig(
+            vocab_size=config.vocab_size,
+            hidden_size=config.hidden_size,
+            num_hidden_layers=config.num_hidden_layers,
+            num_attention_heads=config.num_attention_heads,
+            intermediate_size=config.hidden_size * config.mlp_ratio,
+            max_position_embeddings=config.max_text_len,
+            hidden_dropout_prob=config.drop_rate,
+            attention_probs_dropout_prob=config.drop_rate,
+        )
+
+        if config.cross_modal_transform_shared:
+            self.cross_modal_text_transform = nn.Linear(config.input_text_embed_size, config.hidden_size)
+            self.cross_modal_image_transform = nn.Linear(config.input_image_embed_size, config.hidden_size)
+        else:
+            self.cross_modal_text_transform = nn.ModuleList(
+                [nn.Linear(config.input_text_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+            self.cross_modal_image_transform = nn.ModuleList(
+                [nn.Linear(config.input_image_embed_size, config.hidden_size) for _ in range(config.num_hidden_layers)]
+            )
+
+        self.token_type_embeddings = nn.Embedding(2, config.hidden_size)
+
+        self.vit_model = BridgeTowerCLIP(
+            config.vit_embed_dim,
+            config.resolution_before,
+            config.vit_layers,
+            config.vit_width,
+            config.vit_patch_size,
+            config.vit_transformer_width,
+            resolution_after=config.image_size,
+            stop_gradient=config.stop_gradient,
+            vit_layernorm_shared=config.vit_layernorm_shared,
+            vit_remove_last=config.vit_remove_last,
+        )
+
+        roberta_config = RobertaConfig.from_pretrained(config.tokenizer)
+        self.text_transformer = RobertaModel(roberta_config)
```

Comment:
In transformers we follow the one model one file design philosophy c.f. [a great article about it here](https://huggingface.co/blog/transformers-design-philosophy). If you need Roberta, the necessary layers should be added with `# Copied from` statements in this file.

Reply:
Handled! Thanks for the feedback.